### PR TITLE
[Tests] parallelize

### PR DESCRIPTION
**Pull request by patrickvonplaten**
This PR improves the testing folder structure making it easier to test individual components of `diffusers`. In a first step, we split all fast tests into:
- Fast tests for PRs / Fast PyTorch Models & Schedulers CPU tests (pull_request) 
and
- Fast tests for PRs / Fast PyTorch Pipeline CPU tests (pull_request) 

which saves us roughly 2 minutes.

In follow-up PRs we should now try to speed up the Pipeline tests even a bit more as they still take 6 minutes to run which is too long:
```
39.09s call     tests/pipelines/text_to_video/test_text_to_video.py::TextToVideoSDPipelineFastTests::test_attention_slicing_forward_pass
15.98s call     tests/pipelines/stable_diffusion_2/test_stable_diffusion_attend_and_excite.py::StableDiffusionAttendAndExcitePipelineFastTests::test_inference_batch_consistent
13.28s call     tests/pipelines/stable_diffusion_2/test_stable_diffusion_attend_and_excite.py::StableDiffusionAttendAndExcitePipelineFastTests::test_num_images_per_prompt
12.43s call     tests/pipelines/stable_diffusion/test_stable_diffusion_pix2pix_zero.py::StableDiffusionPix2PixZeroPipelineFastTests::test_inference_batch_consistent
10.44s call     tests/pipelines/text_to_video/test_text_to_video.py::TextToVideoSDPipelineFastTests::test_save_load_optional_components
10.44s call     tests/pipelines/text_to_video/test_text_to_video.py::TextToVideoSDPipelineFastTests::test_save_load_local
9.65s call     tests/pipelines/text_to_video/test_text_to_video.py::TextToVideoSDPipelineFastTests::test_dict_tuple_outputs_equivalent
9.57s call     tests/pipelines/text_to_video/test_text_to_video.py::TextToVideoSDPipelineFastTests::test_progress_bar
7.44s call     tests/pipelines/test_pipelines.py::DownloadTests::test_local_save_load_index
6.99s call     tests/pipelines/test_pipelines.py::DownloadTests::test_download_broken_variant
6.76s call     tests/pipelines/unclip/test_unclip_image_variation.py::UnCLIPImageVariationPipelineFastTests::test_inference_batch_consistent
6.52s call     tests/pipelines/stable_diffusion/test_stable_diffusion_controlnet.py::StableDiffusionMultiControlNetPipelineFastTests::test_inference_batch_consistent
6.31s call     tests/pipelines/unclip/test_unclip.py::UnCLIPPipelineFastTests::test_inference_batch_consistent
6.23s call     tests/pipelines/stable_diffusion_2/test_stable_diffusion_inpaint.py::StableDiffusion2InpaintPipelineFastTests::test_inference_batch_consistent
6.13s call     tests/pipelines/stable_diffusion/test_stable_diffusion_panorama.py::StableDiffusionPanoramaPipelineFastTests::test_num_images_per_prompt
5.84s call     tests/pipelines/text_to_video/test_text_to_video.py::TextToVideoSDPipelineFastTests::test_text_to_video_default_case
5.80s call     tests/pipelines/stable_unclip/test_stable_unclip_img2img.py::StableUnCLIPImg2ImgPipelineFastTests::test_inference_batch_consistent
5.63s call     tests/pipelines/stable_diffusion/test_stable_diffusion_controlnet.py::StableDiffusionControlNetPipelineFastTests::test_inference_batch_consistent
5.51s call     tests/pipelines/stable_diffusion_2/test_stable_diffusion_attend_and_excite.py::StableDiffusionAttendAndExcitePipelineFastTests::test_inference_batch_single_identical
5.49s call     tests/pipelines/stable_unclip/test_stable_unclip.py::StableUnCLIPPipelineFastTests::test_inference_batch_consistent
5.17s call     tests/pipelines/repaint/test_repaint.py::RepaintPipelineFastTests::test_inference_batch_consistent
5.04s call     tests/pipelines/stable_diffusion/test_stable_diffusion_inpaint.py::StableDiffusionInpaintPipelineFastTests::test_inference_batch_consistent
```

Very soon we should also look into @sgugger's amazing test fetcher in `transformers` to skip tests that are unnecessarily run: https://github.com/huggingface/transformers/blob/main/utils/tests_fetcher.py

### CODE REVIEW

File: .github/workflows/pr_tests.yml [hunk 1]
```diff
@@ -71,21 +76,29 @@ jobs:
...
-          tests/
+          tests/pipelines
+
+    - name: Run fast PyTorch Model Scheduler CPU tests
+      if: ${{ matrix.config.framework == 'pytorch_models' }}
+      run: |
+        python -m pytest -n 2 --max-worker-restart=0 --dist=loadfile \
+          -s -v -k "not Flax and not Onnx" \
+          --make-reports=tests_${{ matrix.config.report }} \
+          tests/models tests/schedulers tests/others
```
Comment (sayakpaul) [inline .github/workflows/pr_tests.yml:93]: These two should now run in parallel, right?

**Review by sayakpaul — APPROVED** [review]
Massive ⚡️

**Review by pcuenca — APPROVED** [review]
Looks great!

File: tests/pipelines/test_pipelines_common.py [hunk 2]
```diff
@@ -85,7 +85,7 @@ def params(self) -> frozenset:
         raise NotImplementedError(
             "You need to set the attribute `params` in the child test class. "
             "`params` are checked for if all values are present in `__call__`'s signature."
-            " You can set `params` using one of the common set of parameters defined in`pipeline_params.py`"
+            " You can set `params` using one of the common set of parameters defined ..pipeline_params.py`"
```
[the remark this answers is not in the document]
Comment (patrickvonplaten) [reply tests/pipelines/test_pipelines_common.py:88]: yes haha thanks!

**Pull request merged**